packages/postgrest/lib/src/postgrest_transform_builder.dart

```diff
@@ -195,24 +160,56 @@ class PostgrestTransformBuilder<T> extends PostgrestBuilder<T, T> {
       newHeaders['Accept'] = 'application/vnd.pgrst.object+json';
     }
 
-    return PostgrestTransformBuilder<T>(_copyWith(
-      maybeSingle: true,
-      headers: newHeaders,
-    ));
+    return PostgrestTransformBuilder(
+      _copyWithType(
+        maybeSingle: true,
+        headers: newHeaders,
+      ),
+    );
   }
 
   /// Retrieves the response as CSV.
+  ///
   /// This will skip object parsing.
   ///
   /// ```dart
-  /// postgrest.from('users').select().csv()
+  /// supabase.from('users').select().csv()
   /// ```
-  PostgrestTransformBuilder<T> csv() {
+  PostgrestTransformBuilder<String> csv() {
     final newHeaders = {..._headers};
     newHeaders['Accept'] = 'text/csv';
 
-    return PostgrestTransformBuilder<T>(_copyWith(
-      headers: newHeaders,
-    ));
+    return PostgrestTransformBuilder(
+      _copyWithType(
+        headers: newHeaders,
+      ),
+    );
+  }
+
+  /// Performs additionally to the [select] a count query.
+  ///
+  /// It's used to retrieve the total number of rows that satisfy the
+  /// query. The value for count respects any filters (e.g. eq, gt), but ignores
+  /// modifiers (e.g. limit, range).
+  ///
+  /// This changes the return type from the data only to a [PostgrestResponse] with the data and the count.
+  ///
+  /// ```dart
+  /// final res = await postgrest
+  ///    .from('users')
+  ///    .select()
+  ///    .count(CountOption.exact);
+  /// final users = res.data;
+  /// int count = res.count;
+  /// ```
+  ResponsePostgrestBuilder<PostgrestResponse<T>, T, T> count(
+      CountOption count) {
+    return ResponsePostgrestBuilder(
+      _copyWithType(count: count),
+    );
+  }
+
+  PostgrestBuilder<void, void, void> head() {
+    return _copyWithType(method: METHOD_HEAD);
```

Comment:
I actually found a reason for head. You can check with e.g. single for the existence of a record, without having it to send it to the client.

Reply:
Yeah, or it could be used to invoke `rpc()` method without returning the data too. I wonder if people actually use it that way though. I'm okay leaving it, but I'm okay to also remove it, and only add it if people ever ask for it. I have a feeling no one ever uses `head` alone. The `head` parameter was introduced along side with the `count` parameter in a case where the use only wanted to retrieve the `count` value.

Reply:
I'haven't seen a use as well, but I would keep it for now, because it doesn't involve any extra work than providing that method.